docs/embedded-setup/README.md

```diff
@@ -0,0 +1,193 @@
+}
+
+async function createDestination(body: Destination) {
+  const workspaceId = body.workspaceId;
+
+  const existingDestinations = await getDestinations(workspaceId);
+  if (existingDestinations.data.length > 0) {
+    throw new Error(`Destination already exists for workspace ${workspaceId}`);
+  }
+
+  const vars = fillTemplateVarsFromBody(body, {
+    name: body.name,
+    workspaceId: workspaceId,
+  });
+
+  const templatedBody = renderTemplate(
+    `destination-${body.configuration.destinationType}.mustache.json`,
+    vars
+  ) as Destination;
+
+  const response = await createAirbyteDestination(templatedBody);
+  return response;
+}
+```
+
+Furthermore, the creation of that common destination can also be templatized. In this example we use a [mustache](https://mustache.github.io/) config template for the configuration of a common S3 destination:
+
+```json
+{
+  "name": "{{{name}}}",
+  "definitionId": "4816b78f-1489-44c1-9060-4b19d5fa9362",
+  "workspaceId": "{{{workspaceId}}}",
+  "configuration": {
+    "destinationType": "s3",
+    "access_key_id": "{{{access_key_id}}}",
+    "secret_access_key": "{{{secret_access_key}}}",
+    "s3_bucket_name": "{{{s3_bucket_name}}}",
+    "s3_bucket_path": "${NAMESPACE}/${STREAM_NAME}/${YEAR}_${MONTH}_${DAY}_${EPOCH}_",
+    "s3_bucket_region": "{{{s3_bucket_region}}}",
+    "format": {
+      "format_type": "Parquet",
+      "page_size_kb": 1024,
+      "block_size_mb": 128,
+      "compression_codec": "UNCOMPRESSED",
+      "dictionary_page_size_kb": 1024,
+      "max_padding_size_mb": 8
+    }
+  }
+}
+```
+
+## Configuration Widget
+
+Airbyte Embedded provides a JavaScript library that enables you to host a modal on your website. This allows secure and easy configuration of your customer’s sources. This library works with `templates` that allow you to pre-configure various parts of the source and connection configuration to your specifications. This also will simplify the setup forms presented to your users.
+
+Sources will be validated at configuration time.
+
+### Updating PartialUserConfigs
+
+A “Teamplate” is how you configure the options that users are allowed to choose when setting up a source + connection pair. Any option can be `required`, `optional` or `hidden`, and you can pre-set values for any option.
+
+TODO
+
+### Configuring the Widget
+
+(⚠️ draft)
+
+```shell
+npm install @airbyte-embedded/airbyte-embedded-modal
+```
+
+```ts
+// import the library
+import {
+  AirbyteEmbeddedModal,
+  getToken,
+} from "@airbyte-embedded/airbyte-embedded-modal";
+
+// get a token (in your backend)
+const token = await getToken(oauthKey, oauthSecret);
+
+// Initialize the modal (in your frontend)
+const modal = new AirbyteEmbeddedModal({
+  token, // from your backend
+  workspaceId,
+  onFlowComplete: (source, mask) => {
+    // create the connection on your backend
+    await createConnection(source, mask);
+  },
+});
+
+await modal.open();
+```
+
+### Creating Connections with Masks
```

Comment:
Before this is in front of users, we should switch this to ConfigTemplate (that's what we landed on in yesterday's standup)